tests/t590-operator.c

```diff
@@ -0,0 +1,107 @@
+
+    for (CeedInt e = 0; e < num_elem; e++) {
+      for (CeedInt d = 0; d < dim; d++) {
+        x_array[num_points_per_elem * (e * dim + d) + 0] = 0.25;
+        x_array[num_points_per_elem * (e * dim + d) + 1] = d == 0 ? -0.25 : 0.25;
+        x_array[num_points_per_elem * (e * dim + d) + 2] = d == 0 ? 0.25 : -0.25;
+        x_array[num_points_per_elem * (e * dim + d) + 3] = 0.25;
+      }
+    }
+    CeedVectorSetArray(x_points, CEED_MEM_HOST, CEED_COPY_VALUES, x_array);
+  }
+  for (CeedInt i = 0; i <= num_elem; i++) ind_x[i] = num_elem + 1 + i * num_points_per_elem;
+  for (CeedInt i = 0; i < num_points; i++) ind_x[num_elem + 1 + i] = i;
+  CeedElemRestrictionCreateAtPoints(ceed, num_elem, num_points, dim, num_points * dim, CEED_MEM_HOST, CEED_USE_POINTER, ind_x,
+                                    &elem_restriction_x_points);
+
+  for (CeedInt e = 0; e < num_elem; e++) {
+    CeedInt elem_xy[3] = {1, 1}, n_d[2] = {0, 0};
+
+    for (CeedInt d = 0; d < dim; d++) n_d[d] = num_elem_1d * (p - 1) + 1;
+    {
+      CeedInt r_e = e;
+
+      for (CeedInt d = 0; d < dim; d++) {
+        elem_xy[d] = r_e % num_elem_1d;
+        r_e /= num_elem_1d;
+      }
+    }
+    CeedInt num_nodes_in_elem = p * p, *elem_nodes = ind_u + e * num_nodes_in_elem;
+
+    for (CeedInt n = 0; n < num_nodes_in_elem; n++) {
+      CeedInt g_node = 0, g_node_stride = 1, r_node = n;
+
+      for (CeedInt d = 0; d < dim; d++) {
+        g_node += (elem_xy[d] * (p - 1) + r_node % p) * g_node_stride;
+        g_node_stride *= n_d[d];
+        r_node /= p;
+      }
+      elem_nodes[n] = g_node;
+    }
+  }
+  CeedElemRestrictionCreate(ceed, num_elem, p * p, 1, 1, num_nodes, CEED_MEM_HOST, CEED_USE_POINTER, ind_u, &elem_restriction_u);
+
+  CeedBasisCreateTensorH1Lagrange(ceed, dim, 1, p, q, CEED_GAUSS, &basis_u);
+
+  CeedQFunctionCreateInterior(ceed, 1, mass, mass_loc, &qf_mass);
+  CeedQFunctionAddInput(qf_mass, "u", 1, CEED_EVAL_INTERP);
+  CeedQFunctionAddOutput(qf_mass, "v", 1, CEED_EVAL_INTERP);
+
+  CeedOperatorCreateAtPoints(ceed, qf_mass, CEED_QFUNCTION_NONE, CEED_QFUNCTION_NONE, &op_mass);
+  CeedOperatorSetField(op_mass, "u", elem_restriction_u, basis_u, CEED_VECTOR_ACTIVE);
+  CeedOperatorSetField(op_mass, "v", elem_restriction_u, basis_u, CEED_VECTOR_ACTIVE);
+  CeedOperatorAtPointsSetPoints(op_mass, elem_restriction_x_points, x_points);
+
+  CeedVectorCreate(ceed, num_nodes, &u);
+  CeedVectorSetValue(u, 1.0);
+  CeedVectorCreate(ceed, num_nodes, &v);
+  CeedOperatorApply(op_mass, u, v, CEED_REQUEST_IMMEDIATE);
+
+  {
+    CeedScalar        sum = 0.0;
+    const CeedScalar *v_array;
+
+    CeedVectorGetArrayRead(v, CEED_MEM_HOST, &v_array);
+    for (CeedInt i = 0; i < num_nodes; i++) sum += v_array[i];
+    CeedVectorRestoreArrayRead(v, &v_array);
+    // Summing 9 reference elements, each 2x2 => 36 sq units area
+    if (fabs(sum - 4.0 * num_elem) > CEED_EPSILON * 5e3) printf("Incorrect area computed, %f != %f\n", sum, 4.0 * num_elem);
```

Comment:
I think two other tests would be helpful here 
1) compute element detJ on the mesh and pass it into the points operator
2) an input or output at points